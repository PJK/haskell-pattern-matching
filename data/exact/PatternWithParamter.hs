module PatternWithParamter where

data MyTrinary = A | B | C

data MyBool
    = True
    | False

data MyMaybe = Sum MyTrinary | Nope

extract :: MyMaybe -> MyBool -> MyTrinary
extract (Sum x) False = x
extract (Sum C) True = x

