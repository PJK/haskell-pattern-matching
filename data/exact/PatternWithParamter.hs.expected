{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            {
                "capD": [
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Sum",
                                [
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "C",
                                            []
                                        ]
                                    }
                                ]
                            ]
                        }
                    ],
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Sum",
                                [
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "C",
                                            []
                                        ]
                                    }
                                ]
                            ]
                        }
                    ]
                ],
                "capU": [
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Sum",
                                [
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "A",
                                            []
                                        ]
                                    }
                                ]
                            ]
                        }
                    ],
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Sum",
                                [
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "B",
                                            []
                                        ]
                                    }
                                ]
                            ]
                        }
                    ],
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Nope",
                                []
                            ]
                        }
                    ]
                ],
                "capC": [
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Sum",
                                [
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "C",
                                            []
                                        ]
                                    }
                                ]
                            ]
                        }
                    ]
                ]
            },
            {
                "capD": [],
                "capU": [
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Sum",
                                [
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "B",
                                            []
                                        ]
                                    }
                                ]
                            ]
                        }
                    ],
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Nope",
                                []
                            ]
                        }
                    ]
                ],
                "capC": [
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Sum",
                                [
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "A",
                                            []
                                        ]
                                    }
                                ]
                            ]
                        }
                    ],
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Sum",
                                [
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "A",
                                            []
                                        ]
                                    }
                                ]
                            ]
                        }
                    ]
                ]
            }
        ]
    ]
}